src/NzbDrone.Core/Notifications/Webhook/WebhookService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Specialized;
+using System.IO;
+using System.Linq;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Processes;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.Tv;
+using NzbDrone.Core.Validation;
+using NzbDrone.Core.Rest;
+using RestSharp;
+using NzbDrone.Common.Exceptron.fastJSON;
+
+namespace NzbDrone.Core.Notifications.Webhook
+{
+    public interface IWebhookService
+    {
+        void OnDownload(Series series, EpisodeFile episodeFile, WebhookSettings settings);
+        void OnRename(Series series, WebhookSettings settings);
+        void OnGrab(string grabMessage, WebhookSettings settings);
+        ValidationFailure Test(WebhookSettings settings);
+    }
+
+    public class WebhookService : IWebhookService
+    {
+        private readonly IProcessProvider _processProvider;
+        private readonly Logger _logger;
+
+        public WebhookService(IProcessProvider processProvider, Logger logger)
+        {
+            _processProvider = processProvider;
+            _logger = logger;
+        }
+
+        public void OnDownload(Series series, EpisodeFile episodeFile, WebhookSettings settings)
+        {
+            NotifyWebhook(new { EventType = "Download", Series = series, EpisodeFile = episodeFile }, settings);
```

Comment:
I would much prefer us having a separate data model we use for webhook payload instead of relying on `Series`, `EpisodeFile` which are considered internal contracts. that way if we add/remove/rename stuff from our internal models the webhook implementations are protected by the mapping.

it would be great if you could create a new class, `WebHookSeries`, `WebHookEpisode` and add whatever fields from `Series`, `Episodes` that you think makes sense to an external application. we want to be selective instead of copying the whole thing over since we can add stuff later but once something is added it can't be removed since it would be considered a breaking change.